frontend/src/pages/journal/journal.tsx

```diff
@@ -1,17 +1,40 @@
-import { BackButton } from '#libs/components/components.js';
+import { BackButton, Note } from '#libs/components/components.js';
 import { getValidClassNames } from '#libs/helpers/helpers.js';
-import { useCallback, useSidebarState } from '#libs/hooks/hooks.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useParams,
+  useSidebarState,
+  useState,
+} from '#libs/hooks/hooks.js';
 
 import { JournalSidebar } from './components/journal-sidebar/journal-sidebar.js';
 import styles from './styles.module.scss';
 
 const Journal: React.FC = () => {
+  const { id } = useParams();
+  const dispatch = useAppDispatch();
+  const [isNoteVisible, setIsNoteVisible] = useState(false);
```

Comment:
also useEffect can be removed
```suggestion
  const hasNoteId = Boolean(id) && selectedJournalEntry;
```